src/validationinterface.h

```diff
@@ -35,6 +36,8 @@ class CValidationInterface {
     virtual void Inventory(const uint256 &hash) {};
     virtual void ResendWalletTransactions(int64_t nBestBlockTime) {};
     virtual void BlockChecked(const CBlock&, const CValidationState&) {};
+    virtual void GetScriptForMining(CScript &script) {};
```

Comment:
I would *not* want to create a block with an unspendable coinbase.

I mean the call should return a boolean, or the resulting script should at least be tested for non-emptiness, and the miner should exit if that is the case.